### PR TITLE
xl2tpd: update to 1.3.10

### PR DESCRIPTION
**Pull request by lucize**
update to official 1.3.10

removed patches
explicit added PKG_BUILD_DEPENDS:=libpcap, an indirect depend included in ppp and needed for pfc

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

Maintainer: @yousong 
Compile/Run tested: x86 LEDE r5040-f98f83eada

### CODE REVIEW

File: net/xl2tpd/patches/110-makefile_dont_build_pfc.patch [hunk 1]
```diff
@@ -1,43 +0,0 @@
...
- 
- $(EXEC): $(OBJS) $(HDRS)
- 	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LDLIBS)
-@@ -124,14 +124,10 @@ $(EXEC): $(OBJS) $(HDRS)
- $(CONTROL_EXEC): $(CONTROL_SRCS)
- 	$(CC) $(CFLAGS) $(LDFLAGS) $(CONTROL_SRCS) -o $@
- 
--pfc:
--	$(CC) $(CFLAGS) -c contrib/pfc.c
--	$(CC) $(LDFLAGS) -o pfc pfc.o -lpcap $(LDLIBS)
```
Comment (yousong) [inline net/xl2tpd/patches/110-makefile_dont_build_pfc.patch]: pfc depends on libpcap and the build can potentially fail if it's not present, so I think this patch is still needed

File: net/xl2tpd/patches/100-makefile_opt_flags.patch [hunk 2]
```diff
@@ -1,12 +0,0 @@
---- a/Makefile
-+++ b/Makefile
-@@ -97,7 +97,8 @@ OSFLAGS+= -DUSE_KERNEL
- 
- IPFLAGS?= -DIP_ALLOCATION
- 
--CFLAGS+= $(DFLAGS) -Os -fno-builtin -Wall -DSANITY $(OSFLAGS) $(IPFLAGS)
-+OFLAGS=-Os
-+CFLAGS+= $(DFLAGS) $(OFLAGS) -fno-builtin -Wall -DSANITY $(OSFLAGS) $(IPFLAGS)
```
Comment (yousong) [inline net/xl2tpd/patches/100-makefile_opt_flags.patch]: Please also remove the overridden of `OFLAGS` in the package Makefile.

File: net/xl2tpd/files/options.xl2tpd [hunk 3]
```diff
@@ -11,3 +9,14 @@ novj
...
+mru 1400
+require-mschap
+require-mschap-v2
+lcp-echo-interval 10
+lcp-echo-failure 5
+idle 1800
+connect-delay 5000
+nodefaultroute
+noipdefault
+proxyarp
```
Comment (yousong) [inline net/xl2tpd/files/options.xl2tpd]: Please add more detailed description in the commit message about the changes of default pppd options file

File: net/xl2tpd/patches/110-makefile_dont_build_pfc.patch (hunk 1 above)
Comment (lucize) [reply net/xl2tpd/patches/110-makefile_dont_build_pfc.patch]: ppp has PKG_BUILD_DEPENDS:=libpcap do you think xl2tpd will be used without ppp ?

File: net/xl2tpd/files/options.xl2tpd (hunk 3 above)
Comment (lucize) [reply net/xl2tpd/files/options.xl2tpd]: added

File: net/xl2tpd/patches/100-makefile_opt_flags.patch (hunk 2 above)
Comment (lucize) [reply net/xl2tpd/patches/100-makefile_opt_flags.patch]: removed

File: net/xl2tpd/patches/110-makefile_dont_build_pfc.patch (hunk 1 above)
Comment (yousong) [reply net/xl2tpd/patches/110-makefile_dont_build_pfc.patch]: I missed this indirect dependency, but I think direct deps should be explicitly specified.  You dont build pfc or if you build it you should tell the build system and the reader what it requires

File: net/xl2tpd/files/options.xl2tpd (hunk 3 above)
Comment (yousong) [reply net/xl2tpd/files/options.xl2tpd]: This change should be in a separate commit.

**Comment by lucize** [comment]
splitted and added explicit dependency

**Comment by yousong** [comment]
Thanks!

**Pull request merged**